proof/crefine/ARM/Fastpath_Equiv.thy

```diff
@@ -553,236 +527,229 @@ lemma fastpath_callKernel_SysCall_corres:
+
+                   apply (rule_tac P="inj (case_bool thread (hd (epQueue send_ep)))"
+                            in monadic_rewrite_gen_asm)
+                   apply (rule monadic_rewrite_trans[OF _ monadic_rewrite_transverse])
+                     apply (rule monadic_rewrite_weaken_flags[where F=False and E=True], simp)
+                     apply (rule isolate_thread_actions_rewrite_bind
+                                 fastpath_isolate_rewrites fastpath_isolatables
+                                 bool.simps setRegister_simple
+                                 setVMRoot_isolatable[THEN thread_actions_isolatableD] setVMRoot_isolatable
+                                 doMachineOp_isolatable[THEN thread_actions_isolatableD] doMachineOp_isolatable
+                                 kernelExitAssertions_isolatable[THEN thread_actions_isolatableD]
+                                 kernelExitAssertions_isolatable
+                                 zipWithM_setRegister_simple
+                                 thread_actions_isolatable_bind
                               | assumption
                               | wp assert_inv)+
-                  apply (rule_tac P="\<lambda>s. ksSchedulerAction s = ResumeCurrentThread
-                                      \<and> tcb_at' thread s"
-                             and F=True and E=False in monadic_rewrite_weaken)
-                  apply simp
-                  apply (rule monadic_rewrite_isolate_final)
-                    apply (simp add: isRight_case_sum cong: list.case_cong)
-                   apply (clarsimp simp: fun_eq_iff if_flip
-                                  cong: if_cong)
-                   apply (drule obj_at_ko_at', clarsimp)
-                   apply (frule get_tcb_state_regs_ko_at')
-                   apply (clarsimp simp: zip_map2 zip_same_conv_map foldl_map
-                                        foldl_fun_upd
-                                        foldr_copy_register_tsrs
-                                        isRight_case_sum
-                                  cong: if_cong)
-                   apply (simp add: upto_enum_def fromEnum_def
-                                   enum_register  toEnum_def
-                                   msgRegisters_unfold
-                              cong: if_cong)
-                   apply (clarsimp split: if_split)
-                   apply (rule ext)
-                   apply (simp add: badgeRegister_def msgInfoRegister_def
-                                   ARM.badgeRegister_def
-                                   ARM.msgInfoRegister_def
-                            split: if_split)
-                  apply simp
-                 apply (wp | simp cong: if_cong bool.case_cong
-                           | rule getCTE_wp' gts_wp' threadGet_wp
-                                 getEndpoint_wp)+
+                   apply (rule_tac P="\<lambda>s. ksSchedulerAction s = ResumeCurrentThread
```

Comment:
I think the indentation got a little bit messed up here

Reply:
Yup. I'll clear it up in a separate pass.